assignment-client/src/avatars/AvatarMixerSlavePool.h

```diff
@@ -34,30 +73,26 @@ class AvatarMixerSlaveThread : public QThread, public AvatarMixerSlave {
     using Lock = std::unique_lock<Mutex>;
 
 public:
-    AvatarMixerSlaveThread(AvatarMixerSlavePool& pool, SlaveSharedData* slaveSharedData) :
+    AvatarMixerSlaveThread(AvatarMixerSlavePoolData& pool, SlaveSharedData* slaveSharedData) :
         AvatarMixerSlave(slaveSharedData), _pool(pool) {};
 
     void run() override final;
+    inline void stop() { _stop = true; }
 
 private:
-    friend class AvatarMixerSlavePool;
-
     void wait();
     void notify(bool stopping);
     bool try_pop(SharedNodePointer& node);
 
-    AvatarMixerSlavePool& _pool;
+    AvatarMixerSlavePoolData& _pool;
     void (AvatarMixerSlave::*_function)(const SharedNodePointer& node) { nullptr };
     bool _stop { false };
 };
 
 // Slave pool for avatar mixers
 //   AvatarMixerSlavePool is not thread-safe! It should be instantiated and used from a single thread.
-class AvatarMixerSlavePool {
-    using Queue = tbb::concurrent_queue<SharedNodePointer>;
-    using Mutex = std::mutex;
+class AvatarMixerSlavePool : private AvatarMixerSlavePoolData {
```

Comment:
I like this approach, I think it’s a lot cleaner and easier to follow.  from a naming perspective, it seems weird for a Pool to be a PoolData; I’d expect a Pool to have a PoolData maybe?  or a different name for PoolData?

Reply:
Heh, suggestions welcome?  I could stick this in as a private field too which could work, would just require variables to be qualified when accessed from the pool